syntax = "proto3";

package tutorial;

service AddressBookService {

rpc AddPerson (Person) returns (Response);
rpc GetAddressBook(Empty) returns (AddressBook);
}

message Person{
string name = 1;
int32 id = 2;
string email = 3;

    enum PhoneType{
        MOBILE = 0;
        HOME = 1;
        WORK = 2;
        }

    message PhoneNumber {
    string number = 1;
    PhoneType type =2;
    }

repeated PhoneNumber phones = 4;

}

message AddressBook{
repeated Person people = 1;
}

message Response {
string message = 1;
bool success = 2;
}

message Empty {}
